### PR TITLE
Fix program id retrieval when sharing

### PR DESCRIPTION
**Pull request by annagav**
### What are the relevant tickets?
Fix https://github.com/mitodl/mitxonline/issues/2056

### Description (What does it do?)
Fixes a bug when getting the program id.

### How can this be tested?
Go to view your program record, click enable sharing, check that sharing request was called with the correct program id.

Screenshot 2024-01-16 at 3 37 15 PM

### CODE REVIEW

File: frontend/public/src/containers/pages/records/LearnerRecordsPage.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ export class LearnerRecordsPage extends React.Component<Props, State> {
   }
 
   getOnlyProgramId() {
```
[the remark this answers is not in the document]
Comment (annagav) [reply frontend/public/src/containers/pages/records/LearnerRecordsPage.js:79]: I also has that question. I think the getOnlyProgramId does the same thing getProgramId without the check for `return this.props.match.params.program.length !== 36`. But we should ask @jkachel for the meaning behind it.

[the remark this answers is not in the document]
Comment (jkachel) [reply frontend/public/src/containers/pages/records/LearnerRecordsPage.js:79]: The URLs are the same if you're viewing for a program or a shared record. The shared records use a UUID, hence 36. Neither of these two functions are named particularly well; `getProgramId` is really a check for a shared program record where `getOnlyProgramId` only returns if there's a program ID (aka an int).

**Review by jkachel — CHANGES_REQUESTED** [review]
Functionality works well but I think these two functions should be renamed to something more representative of what they do. These should only be used here so that should be a quick refactor.

**Review by collinpreston — CHANGES_REQUESTED** [review]
So I have a program with 2 required courses.  I also have a program certificate.  When I load the program record page I am seeing an error in the console and the program sharing buttons are not displayed.  When I switch back to the main branch, this does not occur.

**Review by collinpreston — APPROVED** [review]
Very good!

**Review by jkachel — APPROVED** [review]
LGTM 👍

**Pull request merged**